File: apps/java-sdk/src/main/java/com/firecrawl/models/AuditMetadata.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Objects;

/**
 * User attribution included with SIEM logging events.
 */
public final class AuditMetadata {
    private final String username;

    public AuditMetadata(String username) {
        this.username = Objects.requireNonNull(username, "username");
    }

    @JsonProperty("username")
    public String getUsername() {
        return username;
    }
}

File: apps/ruby-sdk/lib/firecrawl/models/audit_metadata.rb
# frozen_string_literal: true

module Firecrawl
  module Models
    # User attribution included with SIEM logging events.
    class AuditMetadata
      attr_reader :username

      def initialize(username:)
        raise ArgumentError, "username must be a string" unless username.is_a?(String)

        @username = username
      end

      def to_h
        { "username" => username }
      end
    end
  end
end
